WcaOnRails/app/helpers/notifications_helper.rb

```diff
@@ -68,13 +68,24 @@ def notifications_for_user(user)
     end
 
     user.delegated_competitions.visible.over
-        .includes(:delegate_report).where(delegate_reports: { posted_at: nil })
-        .each do |competition|
+      .each do |competition|
+        if !competition.delegate_report.posted_at && !competition.results_posted_at?
```

Comment:
We have some nice boolean methods for this: `!competition.delegate_report.posted?` and `!competition.results_posted?`.